src/main/drivers/system.c

```diff
@@ -79,7 +88,19 @@ uint32_t micros(void)
          */
         asm volatile("\tnop\n");
     } while (ms != sysTickUptime);
-    return (ms * 1000) + (usTicks * 1000 - cycle_cnt) / usTicks;
+
+    // SysTick interrupts never get delivered above if called from ISRs.
+    // Treat COUNTFLAG like a pending interrupt flag.
+
+    if (SysTick->CTRL & SysTick_CTRL_COUNTFLAG_Msk) {
```

Comment:
> Writing separate interruptible and non-interuptibe version will help a lot ...

Yeah... I think it will end up with ATOMIC_BLOCK'ing this part anyway if I continue this direction.